src/arrayMethodSort.js

```diff
@@ -4,8 +4,39 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction) {
+    const defaultCompare = (a, b) => {
+      const aStr = String(a);
+      const bStr = String(b);
+
+      for (let i = 0; i < Math.max(aStr.length, bStr.length); i++) {
+        const aChar = aStr.charCodeAt(i) || 0;
+        const bChar = bStr.charCodeAt(i) || 0;
+
+        if (aChar !== bChar) {
+          return aChar - bChar;
+        }
+      }
+
+      return 0;
+    };
+
+    const compare = compareFunction || defaultCompare;
+
+    for (let i = 0; i < this.length; i++) {
+      for (let j = 0; j < this.length - i - 1; j++) {
+        const comparison = compare(this[j], this[j + 1]);
+
+        if (comparison > 0) {
+          const temp = this[j];
+
+          this[j] = this[j + 1];
+          this[j + 1] = temp;
+        }
+      }
+    }
```

Comment:
The bubble sort algorithm used here is correct, but it's known for its inefficiency in larger datasets due to its O(n^2) time complexity. For better performance, especially on larger arrays, consider implementing a more efficient sorting algorithm like quicksort or mergesort.